### PR TITLE
First tomography protocols

### PR DESCRIPTION
**Pull request by EstrellaFG**
- **Protocol_apply_alignmet_subtomo**: protocol to apply transformation matrix to subtomograms (analogous to protocol_apply_alignment for SPA)
- **Protocol_undo_alignment_subtomo**: protocol to undo the alignment
- **Protocol_map_back**: protocol to map back a reference subtomogram to the original tomogram in the coordinates and orientation where subtomograms were extracted. This protocol uses a new xmipp program implemented in: https://github.com/I2PC/xmipp/pull/201#issue-331596894

- **Protocols.conf**: new menu for tomography
- **Convert**: method *getXmippAttribute()* fixed
- **Crop/resize volumes**: modify to manage big volumes (as tomograms)

### CODE REVIEW

File: xmipp3/protocols/protocol_subtomo_map_back.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+        if self.paintingType.get() == 0:
+            painting = 'copy'
+        elif self.paintingType.get() == 1:
+            painting = 'avg %d' % self.threshold.get()
+        elif self.paintingType.get() == 2:
+            painting = 'highlight %d' % self.constant.get()
+        elif self.paintingType.get() == 3:
+            painting = 'copy_binary %d' % self.threshold.get()
+        args = " -i %s -o %s --geom %s --ref %s --method %s" % (self._getExtraPath("tomogram.mrc"),
```
Comment (pconesa) [inline xmipp3/protocols/protocol_subtomo_map_back.py]: I'd create a constant for those strings that mean the same like: "tomogram.mrc". Or use the fileDict functionalty.

File: xmipp3/protocols/protocol_subtomo_map_back.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        self._defineOutputs(outputTomogram=outputTomo)
+        self._defineSourceRelation(self.inputTomogram, outputTomo)
+        self._defineSourceRelation(self.inputClasses, outputTomo)
+
+    #--------------------------- INFO functions --------------------------------
+    def _validate(self):
+        validateMsgs = []
+        # if self.inputCoordinates is None:
+        for subtomo in  self.inputClasses.get().getFirstItem().iterItems():
+            if not subtomo.hasCoordinate3D():
```
Comment (pconesa) [inline xmipp3/protocols/protocol_subtomo_map_back.py]: Does the SetOfClassesSubTomograms have alignment info...If so, we could force the alignment at selection time....and avoiding, maybe this here.

File: xmipp3/protocols/protocol_undo_alignment_subtomo.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                             itemDataIterator=md.iterRows(outputMd, sortByLabel=md.MDL_ITEM_ID))
+        self._defineOutputs(outputSubtomograms=unAlignedSet)
+        self._defineSourceRelation(self.alignedSubtomograms, unAlignedSet)
+        self._defineOutputs(outputSubtomograms=unAlignedSet)
+        self._defineSourceRelation(self.matrixSubtomograms, unAlignedSet)
+
+    # --------------------------- INFO functions --------------------------------------------
+    def _validate(self):
+        validateMsgs = []
+        for subtomo in  self.matrixSubtomograms.get().iterItems():
```
Comment (pconesa) [inline xmipp3/protocols/protocol_undo_alignment_subtomo.py]: Same here? can we force the selection of sets with transformation matrices?

File: xmipp3/protocols.conf [hunk 4]
```diff
@@ -11,8 +11,7 @@ Protocols SPA = [
 	{"tag": "section", "text": "Micrographs", "children": [
 		{"tag": "protocol", "value": "XmippProtPreprocessMicrographs", "text": "default"},
 		{"tag": "protocol_group", "text": "CTF estimation", "openItem": "False", "children": [
-            {"tag": "protocol", "value": "XmippProtCTFMicrographs", "text": "default"},
-            {"tag": "protocol", "value": "XmippProtCTFConsensus", "text": "default"}
+            {"tag": "protocol", "value": "XmippProtCTFMicrographs", "text": "default"}
```
[the remark this answers is not in the document]
Comment (EstrellaFG) [reply xmipp3/protocols.conf]: Fixed here https://github.com/I2PC/scipion-em-xmipp/pull/151/commits/d5e29c3a08827db7151f3129fc8453c9d2e10ed7

File: xmipp3/protocols/protocol_subtomo_map_back.py (hunk 2 above)
Comment (EstrellaFG) [reply xmipp3/protocols/protocol_subtomo_map_back.py]: No, SetOfClassesSubTomograms does not have it.

File: xmipp3/protocols/protocol_undo_alignment_subtomo.py (hunk 3 above)
Comment (EstrellaFG) [reply xmipp3/protocols/protocol_undo_alignment_subtomo.py]: No, the set does not have transformation info.

File: xmipp3/protocols/protocol_subtomo_map_back.py [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+        if self.paintingType.get() == 0 or self.paintingType.get() == 3:
+            if self.removeBackground.get() == True:
+                self.runJob("xmipp_image_operate"," -i %s  --mult 0" % fnTomo)
+
+    def runMapBack(self, classId):
+        TsSubtomo = self.inputClasses.get().getSamplingRate()
+        TsTomo = self.inputTomogram.get().getSamplingRate()
+        scaleFactor = TsSubtomo/TsTomo
+        mdGeometry = xmippLib.MetaData()
+        # if self.inputCoordinates is not None: # Assuming subtomos and coords keep the same order... (which is assume to much...)
```
[the remark this answers is not in the document]
Comment (EstrellaFG) [reply xmipp3/protocols/protocol_subtomo_map_back.py]: Removed in https://github.com/I2PC/scipion-em-xmipp/pull/151/commits/ab5037fbfe6b410425a695b599ecf033c17c0522

**Pull request merged**